.github/workflows/test_on_push.yml

```diff
@@ -95,28 +91,26 @@ jobs:
 
       - name: Run unit tests for GNU/Linux with Python 3.9 and generate coverage report
         if: matrix.os == 'ubuntu-latest' && matrix.python-version == 3.9
-        run: tox -e coverage
+        run: |
+          python -m tox -e unit
+          python -m tox -e coverage
 
       - name: Upload coverage report
         if: matrix.os == 'ubuntu-latest' && matrix.python-version == 3.9
         uses: codecov/codecov-action@v2.1.0
 
-      - name: Run integration tests for GNU/Linux
-        if: matrix.os == 'ubuntu-latest'
+      - name: Run integration tests for GNU/Linux with Python 3.11
+        if: matrix.os == 'ubuntu-latest' && matrix.python-version == 3.11
         run: python -m tox -e integration
 
-      - name: Run unit tests for Windows and MacOS
-        if: matrix.os != 'ubuntu-latest'
+      - name: Run unit tests for Windows and MacOS with Python 3.8, 3.9, and 3.10
+        if: matrix.os != 'ubuntu-latest' && matrix.python-version != 3.11
```

Comment:
Why are we not running unit tests for python 3.11 on Windows and Mac?

Reply:
According to https://github.com/pybamm-team/PyBaMM/issues/2725#issuecomment-1498202987, I ran unit + integration tests on Linux for 3.11, and unit tests only for 3.8 – 3.10 for macOS and Windows; so yes, maybe I should include 3.11 too since it is not being tested at all

Reply:
Yes, I think it should be included too.